### PR TITLE
madness: update to 2024.06.27

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

Update

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: science/madness/Portfile [hunk 1]
```diff
@@ -20,9 +20,9 @@ long_description    MADNESS provides a high-level environment for the solution \
...
                     based on multi-resolution analysis and novel separated representations.
-checksums           rmd160  d9241d1bc207e12ea857938cf1ac111ce2acba83 \
-                    sha256  bbef0cacd92c5bbf546212914d11c269a2e6224078a2f67e43e2a58b5e2803df \
-                    size    36799675
+checksums           rmd160  1689838da6e097d2bd4df1d1e8f7f10382b0868b \
+                    sha256  3b7bea52d12efb74311a3744a23431a836378d1f0762ef79337418530307c5b0 \
+                    size    36804651
 github.tarball_from archive
 
 set py_ver          3.11
```
Comment (herbygillot) [inline science/madness/Portfile:28]: Is 3.12 possible here?

Comment (barracuda156) [reply science/madness/Portfile:28]: Undesirable, but I am kinda tired of arguing over pythons, and for non-essential ports it is easier to just give up and change it to whatever LOL

Comment (herbygillot) [reply science/madness/Portfile]: Ah OK, I always forget that there are issues with Python versions in the PPC/older macOS world.

Comment (barracuda156) [reply science/madness/Portfile]: No big deal here, let it be on 3.12.

**Pull request merged**